File: ver_cpp/T17.cpp
// https://leetcode.cn/problems/missing-number/
#include <bits/stdc++.h>
using namespace std;

class Solution
{
public:
    int missingNumber(vector<int> &nums)
    {
        int n = nums.size();
        int sum = n * (n + 1) / 2;
        for (int i = 0; i < n; ++i)
            sum -= nums[i];
        return sum;
    }
    
};

int main()
{
    Solution s;
    int i;
    vector<int> nums;
    while (cin >> i) nums.push_back(i);
    cout << s.missingNumber(nums) << endl;
    return 0;
}

File: ver_py/T17.py
# -*- coding: utf-8 -*-
# https://leetcode.cn/problems/missing-number/
from typing import List

class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        nums.sort()
        for i, num in enumerate(nums):
            if num != i:
                return i
        return len(nums)

if __name__ == "__main__":
    s = Solution()
    nums = [int(x) for x in input().split()]
    print(s.missingNumber(nums))